diff --git a/compiler/libpc300/libpawnc.c b/compiler/libpc300/libpawnc.c
index c1d1f59..9c6ba28 100755
--- a/compiler/libpc300/libpawnc.c
+++ b/compiler/libpc300/libpawnc.c
@@ -74,10 +74,9 @@ int pc_printf(const char *message,...)
 extern int __attribute__((visibility("default"))) pc_printf(const char *message,...)
 #endif
 #else
-int pc_printf(const char *message, ...)
+extern int __attribute__((visibility("default"))) pc_printf(const char *message, ...)
 #endif
 {
-#if PAWN_CELL_SIZE==32
   int ret;
   va_list argptr;
 
@@ -86,9 +85,6 @@ int pc_printf(const char *message, ...)
   va_end(argptr);
 
   return ret;
-#else
-  return 1;
-#endif
 }
 
 /* pc_error()
@@ -107,7 +103,6 @@ int pc_printf(const char *message, ...)
  */
 int pc_error(int number,char *message,char *filename,int firstline,int lastline,va_list argptr)
 {
-#if PAWN_CELL_SIZE==32
 static char *prefix[3]={ "error", "fatal error", "warning" };
 
   if (number!=0) {
@@ -128,7 +123,6 @@ static char *prefix[3]={ "error", "fatal error", "warning" };
       pc_printf("%s(%d) : %s %03d: ",filename,lastline,pre,number);
   } /* if */
   vprintf(message,argptr);
-#endif
   return 0;
 }